src/ssl/helper.cc

```diff
@@ -136,27 +136,41 @@ void Ssl::Helper::Submit(CrtdMessage const & message, HLPCB * callback, void * d
     GeneratorRequest *request = new GeneratorRequest;
     request->query = rawMessage;
     request->emplace(callback, data);
-    TheGeneratorRequests.emplace(request->query, request);
     debugs(83, 5, "request from " << data << " as " << *request);
-    // ssl_crtd becomes nil if Squid is reconfigured without SslBump or
-    // certificate generation disabled in the new configuration
-    if (ssl_crtd && ssl_crtd->trySubmit(request->query.c_str(), HandleGeneratorReply, request))
+
+    ssl_crtd->generatorRequests.emplace(request->query, request);
+    if (ssl_crtd->trySubmit(request->query.c_str(), HandleGeneratorReply, request))
         return;
 
     ::Helper::Reply failReply(::Helper::BrokenHelper);
     failReply.notes.add("message", "error 45 Temporary network problem, please retry later");
     HandleGeneratorReply(request, failReply);
 }
 
+void
+Ssl::Helper::callBack(::Helper::Xaction &r)
+{
+    const auto callback = r.request.callback;
+    Assure(callback);
+    r.request.callback = nullptr;
+
+    void *cbdata = nullptr;
+    // locked by ::Helper::Xaction::request constructor
+    const auto valid = cbdataReferenceValidDone(r.request.data, &cbdata);
```

Comment:
I am not sure that we need this unlocking here, since Helper::Xaction::request locks and unlocks r.request.data anyway. If we keep it, we should probably adjust Helper::Client::callBack() as well, asserting on cbdataReferenceValidDone() validity (since these Helper::Xaction and Helper::Request objects are created/owned by Helper::Client).